optimum/habana/diffusers/models/unet_spatio_temporal_condition.py

```diff
@@ -0,0 +1,155 @@
+        encoder_hidden_states (`torch.FloatTensor`):
+            The encoder hidden states with shape `(batch, sequence_length, cross_attention_dim)`.
+        added_time_ids: (`torch.FloatTensor`):
+            The additional time ids with shape `(batch, num_additional_ids)`. These are encoded with sinusoidal
+            embeddings and added to the time embeddings.
+        return_dict (`bool`, *optional*, defaults to `True`):
+            Whether or not to return a [`~models.unet_slatio_temporal.UNetSpatioTemporalConditionOutput`] instead of a plain
+            tuple.
+    Returns:
+        [`~models.unet_slatio_temporal.UNetSpatioTemporalConditionOutput`] or `tuple`:
+            If `return_dict` is True, an [`~models.unet_slatio_temporal.UNetSpatioTemporalConditionOutput`] is returned, otherwise
+            a `tuple` is returned where the first element is the sample tensor.
+    """
+    # 1. time
+    timesteps = timestep
+    if not torch.is_tensor(timesteps):
+        # TODO: this requires sync between CPU and GPU. So try to pass timesteps as tensors if you can
+        # This would be a good case for the `match` statement (Python 3.10+)
+        is_mps = sample.device.type == "mps"
+        if isinstance(timestep, float):
+            dtype = torch.float32 if is_mps else torch.float64
+        else:
+            dtype = torch.int32 if is_mps else torch.int64
+        timesteps = torch.tensor([timesteps], dtype=dtype, device=sample.device)
+    elif len(timesteps.shape) == 0:
+        timesteps = timesteps[None].to(sample.device)
+
+    # broadcast to batch dimension in a way that's compatible with ONNX/Core ML
+    batch_size, num_frames = sample.shape[:2]
+    timesteps = timesteps.expand(batch_size)
+
+    t_emb = self.time_proj(timesteps)
+
+    # `Timesteps` does not contain any weights and will always return f32 tensors
+    # but time_embedding might actually be running in fp16. so we need to cast here.
+    # there might be better ways to encapsulate this.
+    t_emb = t_emb.to(dtype=sample.dtype)
+
+    emb = self.time_embedding(t_emb)
+
+    time_embeds = self.add_time_proj(added_time_ids.flatten())
+    time_embeds = time_embeds.reshape((batch_size, -1))
+    time_embeds = time_embeds.to(emb.dtype)
+    aug_emb = self.add_embedding(time_embeds)
+    emb = emb + aug_emb
+
+    # Flatten the batch and frames dimensions
+    # sample: [batch, frames, channels, height, width] -> [batch * frames, channels, height, width]
+    sample = sample.flatten(0, 1)
+    # Repeat the embeddings num_video_frames times
+    # emb: [batch, channels] -> [batch * frames, channels]
+    emb = emb.repeat_interleave(num_frames, dim=0)
+    # encoder_hidden_states: [batch, 1, channels] -> [batch * frames, 1, channels]
+    encoder_hidden_states = encoder_hidden_states.repeat_interleave(num_frames, dim=0)
+
+    # 2. pre-process
+    import habana_frameworks.torch.hpu as hthpu
+
+    # Workaround for SynapseAI 1.11 for Torch Autocast
```

Comment:
Can you please give some details what's the autocast issue here?

Reply:
@yeonsily I copied same WA from https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/diffusers/models/unet_2d_condition.py#L204

Not sure if its still needed for 1.14,1.15 version (probably not), I will check

Reply:
1.11 WA not needed for 1.15.  I removed modified forward unet call from PR as its obsolete